packages/migration/migrations/hearth/20230602120944-populate-valid-locationid.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+
+import {
+  COLLECTION_NAMES,
+  getCollectionDocuments
+} from '../../utils/hearth-helper.js'
+
+import {
+  updateComposition,
+  searchCompositionByCriteria
+  // eslint-disable-next-line import/no-relative-parent-imports
+} from '../../utils/elasticsearch-helper.js'
+
+// THIS MIGRATION POPULATES THE MISSING EVENTLOCATIONIDS OR EVENTJURISDICTIONS IDS DUE TO THE ISSUE - #5242
+
+export const up = async (db, client) => {
+  const session = client.startSession()
+  const limit = 10
+  let processedDocCount = 0
+  const searchCriteria = {
+    bool: {
+      must: [
+        {
+          terms: {
+            type: ['issued', 'certified']
+          }
+        },
+        {
+          bool: {
+            must_not: [
+              {
+                exists: {
+                  field: 'eventLocationId'
+                }
+              },
+              {
+                exists: {
+                  field: 'eventJurisdictionIds'
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+  try {
+    const compositionsWithoutLocationIdsResult =
+      await searchCompositionByCriteria(searchCriteria)
+    const totalCompositionsWithoutLocationIds =
+      compositionsWithoutLocationIdsResult.body.hits.total.value || 0
+
+    while (processedDocCount < totalCompositionsWithoutLocationIds) {
+      const elasticDocBatchResult = await searchCompositionByCriteria(
+        searchCriteria,
+        { size: 10 }
+      )
+      const elasticDocBatch = elasticDocBatchResult.body.hits.hits
+      // eslint-disable-next-line no-console
+      console.log(
+        `Migration - ElasticSearch :: Processing ${processedDocCount + 1} - ${
+          processedDocCount + limit
+        } of ${totalCompositionsWithoutLocationIds} compositions...`
+      )
+
+      for (const elasticDoc of elasticDocBatch) {
+        // populate the missing eventCountry and eventLocationId/eventJurisdictionIds for each doc
+        await updateEventLocationIdOrJurisdictionIds(db, elasticDoc)
+      }
```

Comment:
could we wrap this in a try catch block as I can see that the `updateEventLocationIdOrJurisdictionIds` can throw an error. In that case, we would only be skipping one declaration instead of skipping all the rest